### ISSUE TITLE
Event created in Outlook is considered synced to after two syncs

### ISSUE THREAD

**Issue by NBajanca** [phw198/OutlookGoogleCalendarSync#1924]
**OGCS Version**: v2.10.0.0
**Edition**: Installed  
**Sync Direction**: O<->G

**Bug Description**  
Every event created in Outlook is considered "synced to" Outlook after two syncs. I've tested with events I created and events I was invited to. 

My Sync Options are the following. At this moment, every event in my Outlook Calendar is private.

![image](https://github.com/user-attachments/assets/c51e3618-f2f7-4728-895a-3f52d20df09d)

I've tried to change to the following. The result was all events were public, even the ones from Google.

![image](https://github.com/user-attachments/assets/3fb7d6ff-f514-450e-8e0a-b11dc97ec182)

**Steps to Reproduce the Issue**  
1. Receive an invite
2. Accept
3. Wait for the next sync - the event appears in Google
4. Wait for the next sync - the event changes to Private and to the Calendar Default colour/category in Outlook

**Any other information**  

I'm not attaching the ``OGcalsync.log`` file completely because it has information I consider private. I found every log that references the event, and I'm adding them here:

```
2024-08-01 16:26:18,790 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [1179] -  Comparing Outlook items to Google events...
2024-08-01 16:26:18,790 INFO  17 OutlookGoogleCalendarSync.Console [296] -  Matching calendar items...
2024-08-01 16:26:29,948 INFO  17 OutlookGoogleCalendarSync.Console [296] -  0 Google calendar entries to be deleted.
2024-08-01 16:26:29,948 INFO  17 OutlookGoogleCalendarSync.Console [296] -  1 Google calendar entries to be created.
2024-08-01 16:26:29,948 INFO  17 OutlookGoogleCalendarSync.Console [296] -  65 calendar entries to be compared.
2024-08-01 16:26:29,950 INFO  17 OutlookGoogleCalendarSync.Console [296] -  Creating 1 Google calendar entries
2024-08-01 16:26:30,219 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [409] -  Processing >> 02/08/2024 15:30 => "Test Event"
2024-08-01 16:26:30,220 DEBUG  17 OutlookGoogleCalendarSync.Console [294] -  02/08/2024 15:30 "Test Event"
2024-08-01 16:26:30,434 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  02/08/2024 15:30 => "Test Event"
2024-08-01 16:26:30,435 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_CalendarID=XXX
2024-08-01 16:26:30,435 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_EntryID=XXX
2024-08-01 16:26:30,435 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_GlobalApptID=XXX
2024-08-01 16:26:30,435 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [490] -  Saving timestamp when OGCS created event.
2024-08-01 16:26:31,144 DEBUG  17 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [532] -  Storing the Google event IDs in Outlook appointment.
2024-08-01 16:26:31,236 DEBUG  17 OutlookGoogleCalendarSync.OutlookOgcs.CustomProperty [440] -  02/08/2024 15:30 => "Test Event"
2024-08-01 16:26:31,244 DEBUG  17 OutlookGoogleCalendarSync.OutlookOgcs.CustomProperty [0] -  DoNotForward=False
2024-08-01 16:26:31,253 DEBUG  17 OutlookGoogleCalendarSync.OutlookOgcs.CustomProperty [0] -  googleCalendarID=XXX
2024-08-01 16:26:31,267 DEBUG  17 OutlookGoogleCalendarSync.OutlookOgcs.CustomProperty [0] -  googleEventID=XXX
2024-08-01 16:26:31,335 INFO  17 OutlookGoogleCalendarSync.Sync.Engine+Calendar [595] -  Done.
```

```
2024-08-01 16:56:59,308 DEBUG  8 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [456] -  Processing >> 02/08/2024 15:30 => "Test Event"
2024-08-01 16:57:00,761 DEBUG  8 OutlookGoogleCalendarSync.Console [294] -  02/08/2024 15:30 "Test Event"
2024-08-01 16:57:00,762 DEBUG  8 OutlookGoogleCalendarSync.Console [294] -  AttributeChangePrivacyolNormal → olPrivateCategory/Colour→ OGCS Gray
2024-08-01 16:57:00,795 DEBUG  8 OutlookGoogleCalendarSync.Console [294] -  2 attributes updated.
2024-08-01 16:57:01,056 DEBUG  8 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [715] -  Saving timestamp when OGCS updated appointment.
```

```
2024-08-01 17:11:22,239 DEBUG  12 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [643] -  Processing >> 02/08/2024 15:30 => "Test Event"
2024-08-01 17:11:22,354 DEBUG  12 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [603] -  Doing a dummy update in order to update the last modified date of single instance
2024-08-01 17:11:22,355 DEBUG  12 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [890] -  Saving timestamp when OGCS updated event.
```

```
2024-08-01 17:26:53,159 DEBUG  15 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [456] -  Processing >> 02/08/2024 15:30 => "Test Event"
2024-08-01 17:26:53,550 DEBUG  15 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [424] -  Doing a dummy update in order to update the last modified date.
2024-08-01 17:26:53,601 DEBUG  15 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [715] -  Saving timestamp when OGCS updated appointment.
```

And the same log repeats every sync.

**Comment by phw198** [comment]
Thanks for reporting, but the guidance is to reproduce issues on the latest alpha release.

:arrow_double_up: Please upgrade to ![latest release](https://img.shields.io/github/release/phw198/OutlookGoogleCalendarSync/all.svg) by going to the `About` tab in OGCS, ticking the `Check for Alpha releases` box and then clicking `Check for Update`. 

Alternatively, [download the latest version directly from the project website](https://phw198.github.io/OutlookGoogleCalendarSync/).

Thanks!

**Comment by NBajanca** [comment]
Hi @phw198, 

Thank you for the fast answer. It took some days, but I've updated it to v2.11.0.0, and the issue remains.

Updated log:

```
2024-08-16 14:17:13,031 INFO  12 OutlookGoogleCalendarSync.Console [366] -  Creating 5 Google calendar entries
2024-08-16 14:17:13,301 DEBUG 12 OutlookGoogleCalendarSync.Google.Calendar [555] -  Processing >> 16/08/2024 11:30 => "Meeting 1"
2024-08-16 14:17:13,302 DEBUG 12 OutlookGoogleCalendarSync.Console [364] -  16/08/2024 11:30 "Meeting 1"
2024-08-16 14:17:13,469 DEBUG 12 OutlookGoogleCalendarSync.Google.CustomProperty [347] -  16/08/2024 11:30 => "Meeting 1"
2024-08-16 14:17:13,472 DEBUG 12 OutlookGoogleCalendarSync.Google.CustomProperty [347] -  outlook_CalendarID=XXX
2024-08-16 14:17:13,472 DEBUG 12 OutlookGoogleCalendarSync.Google.CustomProperty [347] -  outlook_EntryID=XXX
2024-08-16 14:17:13,472 DEBUG 12 OutlookGoogleCalendarSync.Google.CustomProperty [347] -  outlook_GlobalApptID=XXX
2024-08-16 14:17:13,475 DEBUG 12 OutlookGoogleCalendarSync.Google.Calendar [636] -  Saving timestamp when OGCS created event.
2024-08-16 14:17:14,106 DEBUG 12 OutlookGoogleCalendarSync.Google.Calendar [681] -  Storing the Google event IDs in Outlook appointment.
2024-08-16 14:17:14,208 DEBUG 12 OutlookGoogleCalendarSync.Outlook.CustomProperty [451] -  16/08/2024 11:30 => "Meeting 1"
2024-08-16 14:17:14,226 DEBUG 12 OutlookGoogleCalendarSync.Outlook.CustomProperty [0] -  DoNotForward=False
2024-08-16 14:17:14,244 DEBUG 12 OutlookGoogleCalendarSync.Outlook.CustomProperty [0] -  googleCalendarID=XXX
2024-08-16 14:17:14,259 DEBUG 12 OutlookGoogleCalendarSync.Outlook.CustomProperty [0] -  googleEventID=XXX
```

```
2024-08-16 14:19:05,875 DEBUG 16 OutlookGoogleCalendarSync.Google.Calendar [806] -  Processing >> 16/08/2024 11:30 => "Meeting 1"
2024-08-16 14:19:05,940 DEBUG 16 OutlookGoogleCalendarSync.Google.Calendar [766] -  Doing a dummy update in order to update the last modified date of single instance
2024-08-16 14:19:05,941 DEBUG 16 OutlookGoogleCalendarSync.Google.Calendar [1080] -  Saving timestamp when OGCS updated event.
```

```
2024-08-16 14:24:45,589 DEBUG 16 OutlookGoogleCalendarSync.Outlook.Calendar [509] -  Processing >> 16/08/2024 11:30 => "Meeting 1"
2024-08-16 14:24:45,878 DEBUG 16 OutlookGoogleCalendarSync.Console [364] -  Privacy: olNormal => olPrivate
2024-08-16 14:24:45,879 DEBUG 16 OutlookGoogleCalendarSync.Console [364] -  Category/Colour:  => OGCS Gray
2024-08-16 14:24:45,879 DEBUG 16 OutlookGoogleCalendarSync.Console [364] -  2 attributes updated.
2024-08-16 14:24:46,142 DEBUG 16 OutlookGoogleCalendarSync.Outlook.Calendar [804] -  Saving timestamp when OGCS updated appointment.
```

Currently, my Outlook calendar has all events set to Private, by OGCS.

I don't know if this is related, but I see another issue: the events created in Google are synced to Outlook. After the second sync, they return to Google, removing the real recipients (replacing by ".ogcs"). This has been happening for months. Meanwhile, I've changed the PCs and installed OGCS fresh without importing any configurations.

```
2024-08-07 22:30:54,562 INFO  3 OutlookGoogleCalendarSync.Console [296] -  Creating 1 Outlook calendar entries
2024-08-07 22:30:54,892 DEBUG  3 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [306] -  Processing >> 25/08/2024 17:00 => "Meeting 2"
2024-08-07 22:30:54,898 DEBUG  3 OutlookGoogleCalendarSync.Console [294] -  25/08/2024 17:00 "Meeting 2"
2024-08-07 22:30:55,413 DEBUG  3 OutlookGoogleCalendarSync.OutlookOgcs.CustomProperty [440] -  25/08/2024 17:00 => "Meeting 2"
2024-08-07 22:30:55,446 DEBUG  3 OutlookGoogleCalendarSync.OutlookOgcs.CustomProperty [0] -  googleCalendarID=XXX
2024-08-07 22:30:55,507 DEBUG  3 OutlookGoogleCalendarSync.OutlookOgcs.CustomProperty [0] -  googleEventID=XXX
2024-08-07 22:30:55,547 DEBUG  3 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [361] -  Saving timestamp when OGCS updated appointment.
2024-08-07 22:30:55,639 DEBUG  3 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [368] -  Storing the Outlook appointment IDs in Google event.
2024-08-07 22:30:55,651 DEBUG  3 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  25/08/2024 17:00 => "Meeting 2"
2024-08-07 22:30:55,651 DEBUG  3 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_CalendarID=XXX
2024-08-07 22:30:55,652 DEBUG  3 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_EntryID=XXX
2024-08-07 22:30:55,652 DEBUG  3 OutlookGoogleCalendarSync.GoogleOgcs.CustomProperty [337] -  outlook_GlobalApptID=XXX
2024-08-07 22:30:55,652 DEBUG  3 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [890] -  Saving timestamp when OGCS updated event.
2024-08-07 22:30:56,112 INFO  3 OutlookGoogleCalendarSync.Sync.Engine+Calendar [704] -  Done.
```

```
2024-08-08 16:40:10,136 DEBUG  8 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [456] -  Processing >> 25/08/2024 17:00 => "Meeting 2"
2024-08-08 16:40:10,336 DEBUG  8 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [424] -  Doing a dummy update in order to update the last modified date.
2024-08-08 16:40:10,395 DEBUG  8 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [715] -  Saving timestamp when OGCS updated appointment.
```

```
2024-08-08 16:54:28,690 DEBUG  8 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [643] -  Processing >> 25/08/2024 17:00 => "Meeting 2"
2024-08-08 16:54:28,929 DEBUG  8 OutlookGoogleCalendarSync.Console [294] -  25/08/2024 17:00 "Meeting 2"
2024-08-08 16:54:28,929 DEBUG  8 OutlookGoogleCalendarSync.Console [294] -  AttributeChangeStart TimezoneEurope/Lisbon → Europe/LondonEnd TimezoneEurope/Lisbon → Europe/LondonAttendee updatedXX1@gmail.com → XX1@gmail.com.ogcsAttendee updatedXX2@gmail.com → XX2@gmail.com.ogcsAttendee XX2@gmail.com - Response Statusaccepted → needsActionRemindernothing → 15
2024-08-08 16:54:28,962 DEBUG  8 OutlookGoogleCalendarSync.Console [294] -  6 attributes updated.
2024-08-08 16:54:28,966 DEBUG  8 OutlookGoogleCalendarSync.GoogleOgcs.Calendar [890] -  Saving timestamp when OGCS updated event.
```
XX1 is my e-mail.
XX2 is the person I invited.

**Comment by phw198** [comment]
I'll be honest, I don't really know what you're reporting - with all the log snippets, you've not included your settings. Nor what you actually want to achieve.

> I've tested with events I created and events I was invited to.

Tested in which calendar? Outlook or Google? From the steps you provided, it sounds like you have new items being created in Outlook, which I'm guessing you don't want their privacy changed? But for items newly created in Google, you want them synced to Outlook as private?

In which case, the option you want is "for items created by sync in Outlook", set to private. Obviously this won't "fix" items that are already private in Outlook which you didn't not want private, you'd have to set them back as public manually.

To respond on your original post:
> My Sync Options are the following. At this moment, every event in my Outlook Calendar is private.

Yes, that would be correct, you've set **all items** synced to Outlook at private - and that's what it's done after it detects a change in Google.

> I've tried to change to the following. The result was all events were public, even the ones from Google.

That needs further explanation - you mean to say that all events both in Google and Outlook had their privacy changed to public?

> I don't know if this is related, but I see another issue: the events created in Google are synced to Outlook. After the second sync, they return to Google, removing the real recipients (replacing by ".ogcs").

If you don't want email cloaking in Google, turn it off?

**Comment by NBajanca** [comment]
Hi, 

Thanks for the long and detailed answer. However, I think you didn't understand the issue (that may be a problem in config). I will try to explain myself better.

Settings: 
![Outlook settings](https://github.com/user-attachments/assets/deda5c6c-6f06-4c46-a02a-d5a8b6582bba)
![Google settings](https://github.com/user-attachments/assets/4e0aa54a-d646-4c51-b381-5738e119d18b)
![Sync options 1/2](https://github.com/user-attachments/assets/57383394-7fd8-437b-95ab-281a437c481a)
![Sync options 2/2](https://github.com/user-attachments/assets/c7cc9007-c9e1-4bfc-aafb-fa7923231f4f)

What I want:
- Events created in Outlook sync to Google with cloaking on.
- Events create in Google sync to Outlook with private.

What I'm getting:
- Events created in Outlook sync to Google with cloaking on. After two more syncs, without any change, the event in Outlook becomes private (as it was syncing from Google)
- Events created in Google sync to Outlook with private. After two more syncs, without any change, the event in Google gets the emails cloaked (as it was syncing from Outlook).

As for your questions:

1. Tested in which calendar?  

Both calendars (Google and Outlook), both options (creating and being invited). 

For example, the event from the last comment was an invited to in my Google Calendar. Besides the organizer and me, now I've also the same emails cloaked (so I got an invite from 1 person, where I was the only one invited, and now the event as 4 emails. The originals and the cloaked) 

2.  you mean to say that all events both in Google and Outlook had their privacy changed to public? 

No, I meant that I was trying the option "for items created" and when I create an event in Google it syncs to Outlook but the privacy stays public.

**Comment by NBajanca** [comment]
Hi @phw198, I stopped being able to replicate the problem. 

I don't know what changed. I changed views. Did many tests. Changed to Google -> Outlook, and back to Outlook <-> Google. Now changing from "all items" to "items created" works as expected :)

The only difference I noticed was the settings started displaying the category name instead of "Calendar Default":

Previously: 

![image](https://github.com/user-attachments/assets/3fb7d6ff-f514-450e-8e0a-b11dc97ec182)

Now:

![image](https://github.com/user-attachments/assets/968f8db1-1822-4c00-8695-fd2d125ab0b5)

If the problem occurs again this week, I will let you know. Otherwise, I will close the issue. Thank you.
